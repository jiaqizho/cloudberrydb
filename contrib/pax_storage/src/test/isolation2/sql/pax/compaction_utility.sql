-- @Description Tests the basic behavior of (lazy) vacuum when called from utility mode
-- 
DROP TABLE IF EXISTS foo;
CREATE TABLE foo (a INT, b INT, c CHAR(128));
CREATE INDEX foo_index ON foo(b);
INSERT INTO foo SELECT i as a, 1 as b, 'hello world' as c FROM generate_series(1, 100) AS i;

DELETE FROM foo WHERE a < 20;
SELECT COUNT(*) FROM foo;
0U: SELECT segment_id, case when pttupcount = 0 then 'zero' when pttupcount <= 5 then 'few' else 'many' end FROM get_pax_aux_table_all('foo');
0U: VACUUM foo;
SELECT COUNT(*) FROM foo;
0U: SELECT segment_id, case when pttupcount = 0 then 'zero' when pttupcount <= 5 then 'few' else 'many' end FROM get_pax_aux_table_all('foo');
