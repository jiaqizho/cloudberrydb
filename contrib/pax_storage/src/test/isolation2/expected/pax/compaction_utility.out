-- @Description Tests the basic behavior of (lazy) vacuum when called from utility mode
--
DROP TABLE IF EXISTS foo;
DROP
CREATE TABLE foo (a INT, b INT, c CHAR(128));
CREATE
CREATE INDEX foo_index ON foo(b);
CREATE
INSERT INTO foo SELECT i as a, 1 as b, 'hello world' as c FROM generate_series(1, 100) AS i;
INSERT 100

DELETE FROM foo WHERE a < 20;
DELETE 19
SELECT COUNT(*) FROM foo;
 count 
-------
 81    
(1 row)
0U: SELECT segment_id, case when pttupcount = 0 then 'zero' when pttupcount <= 5 then 'few' else 'many' end FROM get_pax_aux_table_all('foo');
 segment_id | case 
------------+------
 0          | many 
(1 row)
0U: VACUUM foo;
VACUUM
SELECT COUNT(*) FROM foo;
 count 
-------
 81    
(1 row)
0U: SELECT segment_id, case when pttupcount = 0 then 'zero' when pttupcount <= 5 then 'few' else 'many' end FROM get_pax_aux_table_all('foo');
 segment_id | case 
------------+------
 0          | many 
(1 row)
